reports-scheduler/src/main/java/com/amazon/opendistroforelasticsearch/reportsscheduler/rest/RestReportsJobAction.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package com.amazon.opendistroforelasticsearch.reportsscheduler.rest;
+
+import static com.amazon.opendistroforelasticsearch.reportsscheduler.common.Constants.BASE_SCHEDULER_URI;
+
+import java.util.List;
+import java.util.Locale;
+
+import org.elasticsearch.client.node.NodeClient;
+import org.elasticsearch.cluster.service.ClusterService;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.rest.BaseRestHandler;
+import org.elasticsearch.rest.BytesRestResponse;
+import org.elasticsearch.rest.RestRequest;
+import org.elasticsearch.rest.RestStatus;
+
+import com.amazon.opendistroforelasticsearch.reportsscheduler.rest.handler.ReportsJobActionHandler;
+import com.google.common.collect.ImmutableList;
+
+public class RestReportsJobAction extends BaseRestHandler {
+  public static final String SCHEDULER_JOB_ACTION = "reports_scheduler_job_action";
+  private final String JOB = "job";
+  private final String JOB_ID = "jobId";
+
+  private final Settings settings;
+  private final ClusterService clusterService;
+
+  public RestReportsJobAction(Settings settings, ClusterService clusterService) {
+    this.settings = settings;
+    this.clusterService = clusterService;
+  }
+
+  @Override
+  public String getName() {
+    return SCHEDULER_JOB_ACTION;
+  }
+
+  @Override
+  public List<Route> routes() {
+    return ImmutableList.of(
+        new Route(
+            RestRequest.Method.POST,
+            String.format(Locale.ROOT, "%s/%s/{%s}", BASE_SCHEDULER_URI, JOB, JOB_ID)),
```

Comment:
direct concatenation is better than using String.format (both for CPU and Memory)

Reply:
Do you mean something like this, `new Route(RestRequest.Method.POST, BASE_SCHEDULER_URI + "/" + JOB + "/" + "{" + JOB_ID + "}")`, I'll say the readability is really bad

Reply:
Besides, I checked other plugin repos in Open Distro. They all defined it the same way. e.g. https://github.com/opendistro-for-elasticsearch/anomaly-detection/blob/92a586718f93a9ab934169cab0aca42c8641f4d9/src/main/java/com/amazon/opendistroforelasticsearch/ad/rest/RestAnomalyDetectorJobAction.java#L103

Reply:
May be extract the url formation to constants `private static final String JOB_ID_URL = BASE_SCHEDULER_URI + "/" + JOB + "/{" + JOB_ID + "}";` and use it in function call.

Reply:
fixed